src/EditorFeatures/CSharp/SplitComment/SplitCommentCommandHandler.CommentSplitter.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Indentation;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using static Microsoft.CodeAnalysis.Formatting.FormattingOptions;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.SplitComment
+{
+    internal partial class SplitCommentCommandHandler
+    {
+        private abstract class CommentSplitter
```

Comment:
i'm not certain i get the purpose of the abstract base class given there is only one impl.

Reply:
The abstract class is only there as I based this on the StringSpliiter logic and wasn't initially sure what implementations would be required. I'll look at removing them when I do the VB Implementation.

Reply:
gotcha!